code/game/sound.dm

```diff
@@ -21,12 +20,21 @@
 	for(var/P in listeners)
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			M.sound_or_datum(turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
 	for(var/P in SSmobs.dead_players_by_zlevel[z])
 		var/mob/M = P
 		if(get_dist(M, turf_source) <= maxdistance)
-			M.playsound_local(turf_source, soundin, vol, vary, frequency, falloff, channel, pressure_affected, S)
+			M.sound_or_datum(turf_source, input, vol, vary, frequency, falloff, channel, pressure_affected)
+
+/mob/proc/sound_or_datum(turf/turf_source, input, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE)
```

Comment:
Why is sound_or_datum a mob proc?
Old proc was purely for sound so only mobs would care but new proc is for text too so objects and stuff could recieve it too like say radios